tiledb/api/c_api/DIRECTORY.md

```diff
@@ -6,6 +6,8 @@ Each section has its own directory, containing two headers, at least one API sou
 * **External header**. This header is included in the top-level `tiledb.h` header. These headers will be included in both C and C++ programs and must compile appropriately.
   * Declares C API functions.
   * Only included in `tiledb.h`. Should not appear elsewhere, including in other external headers.
+    * Any non-TileDB header includes *must* happen in `tiledb.h` or `tiledb_experimental.h` *before* the class-specific
+      header `#include "tiledb/api/c_api/...` in order for proper header filtering in wrapper generators (ex: SWIG for TileDB-Java).
```

Comment:
This comment should go in somewhere, but not in this file. It properly belongs in `C_API_STRUCTURE.md`.

I'd make a new section at the end of that document `## Implementation Restrictions`. Since SWIG is the issue, it should be called out as the issue. It's possible to relax the restriction by rewriting the SWIG script, which isn't usually going to be worth it.

Reply:
Updated, thanks.